src/app/InteractionModelEngine.cpp

```diff
@@ -1638,6 +1638,23 @@ void InteractionModelEngine::ResumeSubscriptionsTimerCallback(System::Layer * ap
     auto * iterator = imEngine->mpSubscriptionResumptionStorage->IterateSubscriptions();
     while (iterator->Next(subscriptionInfo))
     {
+        // If subscription happens between reboot and this timer callback, it's already live and should skip resumption
+        bool subscriptionAlreadyLive = false;
+        imEngine->mReadHandlers.ForEachActiveObject([&](ReadHandler * handler) {
+            SubscriptionId subscriptionId;
+            handler->GetSubscriptionId(subscriptionId);
+            if (subscriptionId == subscriptionInfo.mSubscriptionId)
+            {
+                subscriptionAlreadyLive = true;
+            }
+            return Loop::Continue;
```

Comment:
Why wouldn't we just `return Loop::Break` here when the id matches?

And if we do that, do we need `subscriptionAlreadyLive`?  We can just test whether `ForEachActiveObject` returns `Loop::Break` to see whether we found a match.

Reply:
Makes sense. Will change